spec/controllers/scim_rails/scim_groups_controller_spec.rb

```diff
@@ -0,0 +1,258 @@
+require 'spec_helper'
+
+RSpec.describe ScimRails::ScimGroupsController, type: :controller do
+  include AuthHelper
+
+  routes { ScimRails::Engine.routes }
+  let(:company) { create(:company) }
+
+  describe 'index' do
+    context 'without authorization' do
+      before { get :index }
+
+      it "returns scim+json content type" do
+        expect(response.content_type).to eq "application/scim+json"
+      end
+
+      it "fails with no credentials" do
+        expect(response.status).to eq 401
+      end
+
+      it "fails with invalid credentials" do
+        request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials("unauthorized","123456")
+        expect(response.status).to eq 401
+      end
+    end
+
+    context 'with authorization' do
+      before :each do
+        http_login(company)
+      end
+
+      let(:user_list_length) { 3 }
+      let(:user_list) { create_list(:user, user_list_length, first_name: Faker::Name.first_name, last_name: Faker::Name.last_name, company: company) }
+      let(:response_body) { JSON.parse(response.body) }
+
+      it 'returns scim+json content type' do
+        get :index
+        expect(response.content_type).to eq "application/scim+json"
+      end
+
+      it 'is successful' do
+        get :index
+        expect(response.status).to eq(200)
+      end
+
+      context 'when less than 100 existing groups' do
+        let(:total_group_count) { 10 }
+
+        let(:returned_resource) { response_body["Resources"].first }
+
+        it 'returns all results' do
+          create_list(:group, total_group_count, users: user_list, company: company)
```

Comment:
move to a let in the context, then you could do the same before block for the request